Islandora/islandora
# Term revisions can now go to Fedora

**GitHub Issue**: #1042 

* Other Relevant Links (Google Groups discussion, related pull requests,
 Release pull requests, etc.)

# What does this Pull Request do?
Lets terms have revisions in Fedora.

# What's new?

Make the media revision-getter generic so that it can handle taxonomy terms too. (Nodes have a special shortcut function and don't need the long workaround to get revision IDs)

# How should this be tested?

Make a taxonomy term. Edit it, and check "create a new revision". 

Without this PR: The term in Fedora changes, but no revisions are created.
With this PR: You get a new revision.

# Documentation Status

* Does this change existing behaviour that's currently documented? no
* Does this change require new pages or sections of documentation? no
* Who does this need to be documented for? n/a
* Associated documentation pull request(s): ___  or documentation issue ___

# Additional Notes:
Any additional information that you think would be helpful when reviewing this
 PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers
